### PR TITLE
Case 21185: protect nitpick from bad filename

### PR DESCRIPTION
**Pull request by NissimHadar**
https://highfidelity.manuscript.com/f/cases/21185/Protect-nitpick-from-bad-file-name

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17259/HighFidelity-Beta-PR14912-56483f4.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17259/HighFidelity-Beta-PR14912-56483f4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17259/HighFidelity-Beta-Interface-PR14912-56483f4.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17268/HighFidelity-Beta-PR14912-be5f9b2.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17268/HighFidelity-Beta-PR14912-be5f9b2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17268/HighFidelity-Beta-Interface-PR14912-be5f9b2.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17270/HighFidelity-Beta-PR14912-301b421.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17270/HighFidelity-Beta-PR14912-301b421.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17270/HighFidelity-Beta-Interface-PR14912-301b421.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17270/HighFidelity-Beta-PR14912-301b421.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17270/HighFidelity-Beta-Interface-PR14912-301b421.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17271/HighFidelity-Beta-PR14912-58c0078.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17271/HighFidelity-Beta-PR14912-58c0078.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17271/HighFidelity-Beta-Interface-PR14912-58c0078.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17271/HighFidelity-Beta-PR14912-58c0078.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17271/HighFidelity-Beta-Interface-PR14912-58c0078.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17273/HighFidelity-Beta-PR14912-701c18c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17273/HighFidelity-Beta-PR14912-701c18c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17273/HighFidelity-Beta-Interface-PR14912-701c18c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17273/HighFidelity-Beta-PR14912-701c18c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17273/HighFidelity-Beta-Interface-PR14912-701c18c.exe

File: tools/nitpick/src/AWSInterface.cpp [hunk 1]
```diff
@@ -48,10 +65,8 @@ void AWSInterface::extractTestFailuresFromZippedFolder() {
     // For a test results zip file called `D:/tt/TestResults--2018-10-02_16-54-11(9426)[DESKTOP-PMKNLSQ].zip`
     //   the folder will be called `TestResults--2018-10-02_16-54-11(9426)[DESKTOP-PMKNLSQ]`
     //   and, this folder will be in the working directory
-    QStringList parts = _testResults.split('/');
-    QString zipFolderName = _workingDirectory + "/" + parts[parts.length() - 1].split('.')[0];
-    if (QDir(zipFolderName).exists()) {
-        QDir dir = zipFolderName;
+    if (QDir(_zipFolderName).exists()) {
+        QDir dir = _zipFolderName;
```
Comment (sabrina-shanman) [inline tools/nitpick/src/AWSInterface.cpp]: Since _zipFolderName is only used here, why not just make it a function parameter?

Comment (NissimHadar) [reply tools/nitpick/src/AWSInterface.cpp]: Fixed.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17279/HighFidelity-Beta-PR14912-6752092.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17279/HighFidelity-Beta-PR14912-6752092.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17279/HighFidelity-Beta-Interface-PR14912-6752092.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17279/HighFidelity-Beta-PR14912-6752092.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17279/HighFidelity-Beta-Interface-PR14912-6752092.exe

**Pull request merged**